apm-agent-core/src/main/java/co/elastic/apm/agent/bci/classloading/IndyPluginClassLoader.java

```diff
@@ -41,9 +43,15 @@ public class IndyPluginClassLoader extends ByteArrayClassLoader.ChildFirst {
 
     public IndyPluginClassLoader(@Nullable ClassLoader targetClassLoader, ClassLoader agentClassLoader, Map<String, byte[]> typeDefinitions) {
         // See getResource on why we're using PersistenceHandler.LATENT over PersistenceHandler.MANIFEST
-        super(getParent(targetClassLoader, agentClassLoader), true, typeDefinitions, PersistenceHandler.LATENT);
+        super(getParent(targetClassLoader, agentClassLoader),
+            true,
+            typeDefinitions,
+            PrivilegedActionUtils.getProtectionDomain(agentClassLoader.getClass()), // inherit protection domain from agent CL
```

Comment:
[for reviewer] this change allows to provide the protection domain that will be used for classes that are loaded through tne `IndyPluginClassLoader`, without that, we get a `null` protection domain for classes that are loaded by this CL, hence making us unable to get the right permissions in advice code, even with `doPrivileged` wrapping.

Reply:
We just looked at this together- the BB class loader makes specific use in this protection domain and the URLClassLoader uses the caller PD if it is not provided with an explicit one, which will be equivalent as passing the caller's PD.